src/Srclib/Types.hs

```diff
@@ -8,18 +8,153 @@ module Srclib.Types (
   SourceUserDefDep (..),
   SourceRemoteDep (..),
   Locator (..),
+  LicenseSourceUnit (..),
+  LicenseScanType (..),
+  LicenseUnit (..),
+  LicenseUnitData (..),
+  LicenseUnitMatchData (..),
   renderLocator,
   parseLocator,
 ) where
 
 import Data.Aeson
+import Data.List.NonEmpty (NonEmpty)
 import Data.Maybe (fromMaybe)
 import Data.String.Conversion (ToText, toText)
 import Data.Text (Text)
 import Data.Text qualified as Text
 import Path (File, SomeBase)
 import Types (GraphBreadth (..))
 
+data LicenseScanType = CliLicenseScanned
+  deriving (Eq, Ord, Show)
+
+instance ToText LicenseScanType where
+  toText :: LicenseScanType -> Text
+  toText _ = "cli-license-scanned"
```

Comment:
[nit]: Simplify and future-proof:

We don't need the type sig, since that's the only type that this function could possibly have, dictated by the type class.

```suggestion
instance ToText LicenseScanType where
  toText CliLicenseScanned = "cli-license-scanned"
```